plugins/adrv9002.c

```diff
@@ -1440,93 +1453,99 @@ static int profile_gen_config_get_from_device(struct adrv9002_config *cfg, gpoin
-					"nco_frequency", ret);
+			// rx.channel_bandwidth_hz
+			ret = iio_channel_attr_read(rx, "rf_bandwidth", buf, sizeof(buf));
+			if(ret < 0) {
+				sprintf(message, "\nFailed to get rx channel: %s attr: %s! error code: %d", chann_str,
+					"rf_bandwidth", ret);
 				goto iio_error;
 			}
-		} else {
-			rx_config[chann].nco_frequency_hz = atoi(buf);
-		}
+			rx_config[chann].channel_bandwidth_hz = atoi(buf);
+
+			// rx.adc_high_performance_mode
+			ret = iio_device_debug_attr_read(priv->adrv9002, chann == 0 ? "rx0_adc_type" : "rx1_adc_type",
+							 buf, sizeof(buf));
+			if(ret < 0) {
+				sprintf(message, "\nFailed to get rx channel: %s attr: %s! error code: %d", chann_str,
+					chann == 0 ? "rx0_adc_type" : "rx1_adc_type", ret);
+				goto iio_error;
+			}
+			rx_config[chann].adc_high_performance_mode = strstr(buf, "HP") != NULL;
+
+			// rx.analog_filter_biquad
+			rx_config[chann].analog_filter_biquad =
+				default_cfg.radio_cfg.rx_config[chann].analog_filter_biquad; // TODO
+
+			// rx.analog_filter_bandwidth_hz
+			rx_config[chann].analog_filter_bandwidth_hz =
+				default_cfg.radio_cfg.rx_config[chann].analog_filter_bandwidth_hz; // TODO
+
+			// rx.nco_enable
+			rx_config[chann].nco_enable = default_cfg.radio_cfg.rx_config[chann].nco_enable; // TODO
+
+			// rx.nco_frequency_hz
+			ret = iio_channel_attr_read(rx, "nco_frequency", buf, sizeof(buf));
+			if(ret < 0) {
+				if(ret == -ENOTSUPP) {
+					rx_config[chann].nco_frequency_hz = 0;
+				} else {
+					sprintf(message, "\nFailed to get channel: %s attr: %s! error code: %d",
+						chann_str, "nco_frequency", ret);
+					goto iio_error;
+				}
+			} else {
+				rx_config[chann].nco_frequency_hz = atoi(buf);
+			}
 
-		// rx.rf_port
-		rx_config[chann].rf_port = default_cfg.radio_cfg.rx_config[chann].rf_port; // TODO
+			// rx.rf_port
+			rx_config[chann].rf_port = default_cfg.radio_cfg.rx_config[chann].rf_port; // TODO
 
+		} else {
+			rx_config[chann].sample_rate_hz = 0;
+			rx_config[chann].frequency_offset_correction_enable = 0;
+			rx_config[chann].analog_filter_power_mode = 0;
+			rx_config[chann].channel_bandwidth_hz = 0;
+			rx_config[chann].adc_high_performance_mode = 0;
+			rx_config[chann].analog_filter_biquad = 0;
+			rx_config[chann].analog_filter_bandwidth_hz = 0;
+			rx_config[chann].nco_enable = 0;
+			rx_config[chann].nco_frequency_hz = 0;
+			rx_config[chann].rf_port = 0;
+		}
 		radio_config.rx_config[chann] = rx_config[chann];
 
 		// tx.orx_enabled
 		ret = iio_channel_attr_read(rx, "orx_en", buf, sizeof(buf));
 		if(ret < 0) {
 			if(ret == -ENODEV) {
-				tx_config[chann].orx_enabled =
+				radio_config.tx_config[chann].orx_enabled =
 					default_cfg.radio_cfg.tx_config[chann].orx_enabled; // Temporary fix
 			} else {
-				sprintf(message, "\nFailed to get channel: %s attr: %s! error code: %d", chann_str,
+				sprintf(message, "\nFailed to get rx channel: %s attr: %s! error code: %d", chann_str,
```

Comment:
Also unrelated change... When are minor changes like this that is ok to put it in the same patch, you should at least mention it in the commit message.